apps/financial-aid/web-veita/src/components/ApplicationsTable/ApplicationsTable.tsx

```diff
@@ -8,50 +8,88 @@ import cn from 'classnames'
 import {
   TableHeaders,
   TableBody,
+  LoadingContainer,
+  TableSkeleton,
 } from '@island.is/financial-aid-web/veita/src/components'
-import { Application } from '@island.is/financial-aid/shared/lib'
+import {
+  Application,
+  ApplicationState,
+  ApplicationStateUrl,
+} from '@island.is/financial-aid/shared/lib'
 import { TableHeadersProps } from '@island.is/financial-aid-web/veita/src/routes/ApplicationsOverview/applicationsOverview'
 
+import { useAllApplications } from '@island.is/financial-aid-web/veita/src/utils/useAllApplications'
+
 interface PageProps {
   applications: Application[]
+  setApplications: React.Dispatch<
+    React.SetStateAction<Application[] | undefined>
+  >
   headers: TableHeadersProps[]
 }
 
-const ApplicationsTable = ({ applications, headers }: PageProps) => {
+const ApplicationsTable = ({
+  applications,
+  headers,
+  setApplications,
+}: PageProps) => {
   const router = useRouter()
 
+  const changeApplicationTable = useAllApplications()
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  const updateApplicationAndTable = async (
+    applicationId: string,
+    state: ApplicationState,
+  ) => {
+    setIsLoading(true)
+    const updateApplicationTable = await changeApplicationTable(
+      applicationId,
+      state,
+      router.pathname.substring(1) as ApplicationStateUrl,
```

Comment:
Somewhat afraid of this, this could cause unexpected effects if we decide to change the url's at any point. We could write a helper function to check what values are contained in the url and that changes `ApplicationStateUrl`. Any thoughts?